packages/forms-web-app/src/controllers/common/enter-code.js

```diff
@@ -107,81 +101,112 @@ async function sendTokenToLpaUser(req) {
 	}
 }
 
-const getEnterCode = (views, appealInSession) => {
+/**
+ * @param {{EMAIL_ADDRESS: string, ENTER_CODE: string, REQUEST_NEW_CODE: string}} views
+ * @param {boolean} [isGeneralLogin]
+ * @returns {import('express').Handler}
+ */
+const getEnterCode = (views, isGeneralLogin) => {
 	return async (req, res) => {
 		const {
 			body: { errors = {} }
 		} = req;
 
-		const action = req.session?.enterCode?.action ?? enterCodeConfig.actions.saveAndReturn;
-
-		if (action === enterCodeConfig.actions.saveAndReturn) {
-			req.session.enterCode = req.session.enterCode || {};
-			req.session.enterCode.action = enterCodeConfig.actions.saveAndReturn;
+		/** @type {string|undefined} */
+		const enterCodeId = req.params.id;
+		if (enterCodeId) {
+			req.session.enterCodeId = enterCodeId;
 		}
 
+		const action = req.session?.enterCode?.action ?? enterCodeConfig.actions.saveAndReturn;
+		const isReturningFromEmail = action === enterCodeConfig.actions.saveAndReturn;
+		const isAppealConfirmation = !isGeneralLogin && !isReturningFromEmail;
+
 		// show new code success message only once
 		const newCode = req.session?.enterCode?.newCode;
 		if (newCode) {
 			delete req.session?.enterCode?.newCode;
 		}
 
-		const url = `/${views.REQUEST_NEW_CODE}`;
-		const confirmEmailUrl =
-			action === enterCodeConfig.actions.confirmEmail ? `/${views.EMAIL_ADDRESS}` : null;
+		if (isGeneralLogin) {
+			if (!(await isFeatureActive(FLAG.ENROL_USERS))) {
+				throw new Error('unhandled journey for GET: enter-code');
+			}
 
-		//this handles old save & return url (without id params)
-		if (appealInSession && req.session.appeal?.id && !req.params.id) {
-			res.redirect(`/${views.ENTER_CODE_URL}/${req.session.appeal.id}`);
-			return;
+			const email = getSessionEmail(req.session, false);
+
+			try {
+				await sendToken(undefined, action, email);
+			} catch (e) {
+				logger.error(e, 'failed to send token to general login user');
```

Comment:
Should this not show an error to the user?

Reply:
Yes I see. I guess I was thinking about "we failed to send the email" type errors - which perhaps we handle differently so they can retry? Possibly for another ticket/PR.